src/frontend/lib/file-system.ts

```diff
@@ -8,26 +8,37 @@ export function convertFileUriToPosixPath(fileUri: string) {
   return fileUri.replace(/^file:\/\//, '');
 }
 
-// TODO: Some overlap with selectFile() from lib/utils but fixes some usage limitations. Ideally use this for everything
+/**
+ * Select a single file from the device file system.
+ *
+ * @param opts.copyToCacheDirectory Copy the selected file to the cache directory for immediate usage. Note that if this is set to `true`, the file can be immediately used and the resulting URI for selected file will be prefixed with `file://` instead of `content://`.
+ * @param opts.extensionFilters File extensions to allow. Leaving this unspecified allows any kind of file to be selected.
+ * @param opts.mimeFilters MIME types to allow. Leaving this unspecified allows any kind of MIME type to be considered.
+ *
+ * @returns The selected file. If `null` it means the selection was cancelled.
+ */
 export async function selectFile({
-  mimeFilters,
+  copyToCacheDirectory,
   extensionFilters,
+  mimeFilters,
 }: {
-  mimeFilters?: Array<string>;
+  copyToCacheDirectory: boolean;
```

Comment:
i made this option required because its value affects the result of the URI that the file has, as documented in the typedoc comment above. makes it slightly clearer about what you can or can't do with the file.

it's kind of an annoying difference that affects us, unfortunately.